### PR TITLE
Feature/error reporting

### PR DESCRIPTION
**Pull request by stevenhillcox**

### CODE REVIEW

File: VotingApplication/VotingApplication.Web/Scripts/Interceptors/ErrorInterceptor.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+                $rootScope.error.readableMessage = firstError[0];
+            } else if (rejection.data && rejection.data.error_description) {
+                // Handle authentication errors
+                $rootScope.error.readableMessage = rejection.data.error_description;
+            } else if (rejection.statusText) {
+                // Handle generic server messages
+                $rootScope.error.readableMessage = rejection.statusText;
+            } else {
+                // Catch all for anything else
+                $rootScope.error.readableMessage = rejection.data ? rejection.data.Message : "An error has occured";
```
Comment (tpkelly) [inline VotingApplication/VotingApplication.Web/Scripts/Interceptors/ErrorInterceptor.js]: Looks like JSLint isn't checking the Interceptors/ directory, but mine flags up the single quotes issue

**Comment by tpkelly** [comment]
Missing Poll for ID is just displaying as "Bad Request". We should add that to the list of "Custom error messages we need" :)

**Comment by JDawes-ScottLogic** [comment]
:+1:

**Comment by tpkelly** [comment]
Fine for this to go through once merge conflicts are resolved

It looks like the "Bad Request" i was seeing was because I dropped a digit from the Guid and it threw an error before reaching our code.

**Pull request merged**